webui/react/src/pages/ResourcePool/NodeElement.tsx

```diff
@@ -0,0 +1,83 @@
+import Tooltip from 'hew/Tooltip';
+import React, { PropsWithChildren, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import { Resource, SlotsRecord } from 'types';
+
+import css from './NodeElement.module.scss';
+
+export type SimplifiedNode = { container?: { id: string } };
+interface NodeElementProps {
+  name: string;
+  resources: Resource[] | SimplifiedNode[];
+  slots?: SlotsRecord;
+  isRunning?: boolean;
+  limitSlots?: number;
+}
+
+const NodeElement: React.FC<PropsWithChildren<NodeElementProps>> = ({
+  name,
+  slots,
+  resources,
+  isRunning = true,
+  limitSlots = 0,
+}) => {
+  const [containerWidth, setContainerWidth] = useState(0);
+  const shouldTruncate = name.length > 5;
+  const slotsContainer = useRef<HTMLSpanElement>(null);
+  const slotsData = useMemo(
+    () => (slots !== undefined ? Object.values(slots) : resources),
+    [slots, resources],
+  );
+  const slotStyles = [css.nodeSlot];
+  const nodeStyles = [css.node];
+  const nodeClusterStyles = [css.nodeCluster];
+
+  if (!isRunning) {
+    nodeStyles.push(css.notRunning);
+    nodeClusterStyles.push(css.notRunning);
+  }
+
+  const getSlotStyles = useCallback(
```

Comment:
i think something is wrong in the logic with multiple slots. Visual is different from [figma](https://www.figma.com/file/4daEeZctbKDbkdvrhgChiN/Fault-Tolerance-Q4-23?type=design&node-id=231-1196&mode=design&t=KJlImMgWGUDJIN4d-0)

<img width="714" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/e858251e-dc37-45f5-b49d-db1938efdefb">

Reply:
i dont remember which one